app/api/routers/breeze_buddy/analytics/__init__.py

```
@@ -107,3 +115,67 @@ async def get_analytics(
+        f"User {current_user.user_id} requesting call recording for call_sid: {call_sid}"
+    )
+
+    try:
+        # Get recording URL and provider from database
+        result = await get_recording_url_and_provider_by_call_id(call_sid)
+
+        if not result:
+            logger.error(f"No recording found for call_sid: {call_sid}")
+            raise HTTPException(
+                status_code=404, detail=f"Recording not found for call_sid: {call_sid}"
+            )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)